coap/service.go

```diff
@@ -0,0 +1,28 @@
+package coap
+
+import (
+	"errors"
+
+	"github.com/mainflux/mainflux"
+)
+
+var (
+	// ErrFailedMessagePublish indicates that message publishing failed.
+	ErrFailedMessagePublish = errors.New("failed to publish message")
+
+	// ErrFailedSubscription indicates that client couldn't subscribe to specified channel.
+	ErrFailedSubscription = errors.New("failed to subscribe to a channel")
+
+	// ErrFailedConnection indicates that service couldn't connect to message broker.
+	ErrFailedConnection = errors.New("failed to connect to message broker")
+)
+
+// Service specifies coap service API.
+type Service interface {
+	mainflux.MessagePublisher
+	// Subscribes to channel with specified id and adds subscription to
+	// service map of subscriptions under given ID.
+	Subscribe(string, string, chan mainflux.RawMessage) error
+	// Unsubscribe method is used to stop observing resource.
+	Unsubscribe(string) error
```

Comment:
Can you avoid `Unsubscribe` method by using channel to unsubscribe?

Reply:
Yes, but I didn't want to enforce that interface where it is not necessary. I already got `Subscription` from NATS client and I've got a map for CoAP purposes, so I didn't want to enforce that kind of unsubscription handling and to make one more goroutine if it isn't really necessary. Instead, I use what I've already got.

Reply:
I have already suggested renaming this function. However, I am not sure whether it should be exported or not. Let's try handling this case via channels (message passing), and renaming it if needed. If that works, we can safely "unexport" it.